ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoiceNotifier.java

```diff
@@ -185,6 +211,8 @@ private void updatePayloadAttributes(
           currentSlot);
       return;
     }
+    this.lastPayloadIdBeaconBlockRoot = recentChainData.getBestBlockRoot();
```

Comment:
This probably isn't correct. We are using the optimistic head to send forkChoiceUpdated, not the chain head.

Reply:
Even with this PR we are failing to create blocks and I suspect this is why but we may want to improve the payload ID not available message to include the root and slot we do actually have:
```
2021-11-23 14:44:48.007+10:00 | forkChoiceNotifier-async-0 | ERROR | ForkChoiceNotifier | PayloadId not available for requested Beacon Block Root 0x2b429d9d8b2ec2a6b54917b995b191baab1bec8cb2be2d164f218978bef4fa70 and Slot 210
2021-11-23 14:44:48.008+10:00 | validator-async-8 | ERROR | teku-validator-log | Validator   *** Failed to produce block  Slot: 210 Validator: abf8641
tech.pegasys.teku.spec.logic.common.statetransition.exceptions.StateTransitionException: tech.pegasys.teku.spec.logic.common.statetransition.exceptions.BlockProcessingException: Execution payload parent hash does not match previous execution payload header
at tech.pegasys.teku.spec.logic.common.util.BlockProposalUtil.createNewUnsignedBlock(BlockProposalUtil.java:90) ~[teku-ethereum-spec-develop.jar:21.11.0+64-gdf02342]
at tech.pegasys.teku.spec.Spec.createNewUnsignedBlock(Spec.java:508) ~[teku-ethereum-spec-develop.jar:21.11.0+64-gdf02342]
at tech.pegasys.teku.validator.coordinator.BlockFactory.createUnsignedBlock(BlockFactory.java:70) ~[teku-beacon-validator-develop.jar:21.11.0+64-gdf02342]
at tech.pegasys.teku.validator.coordinator.ValidatorApiHandler.createBlock(ValidatorApiHandler.java:305) ~[teku-beacon-validator-develop.jar:21.11.0+64-gdf02342]
at tech.pegasys.teku.validator.coordinator.ValidatorApiHandler.lambda$createUnsignedBlock$12(ValidatorApiHandler.java:288) ~[teku-beacon-validator-develop.jar:21.11.0+64-gdf02342]
at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$thenApplyChecked$29(SafeFuture.java:422) ~[teku-infrastructure-async-develop.jar:21.11.0+64-gdf02342]
at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1146) ~[?:?]
at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2137) ~[?:?]
at tech.pegasys.teku.dataproviders.generators.CachingTaskQueue.lambda$completePendingTask$9(CachingTaskQueue.java:199) ~[teku-ethereum-dataproviders-develop.jar:21.11.0+64-gdf02342]
at tech.pegasys.teku.infrastructure.async.SafeFuture.fromRunnable(SafeFuture.java:154) ~[teku-infrastructure-async-develop.jar:21.11.0+64-gdf02342]
at tech.pegasys.teku.infrastructure.async.AsyncRunner.lambda$runAsync$0(AsyncRunner.java:23) ~[teku-infrastructure-async-develop.jar:21.11.0+64-gdf02342]
at tech.pegasys.teku.infrastructure.async.SafeFuture.of(SafeFuture.java:73) ~[teku-infrastructure-async-develop.jar:21.11.0+64-gdf02342]
at tech.pegasys.teku.infrastructure.async.ScheduledExecutorAsyncRunner.lambda$createRunnableForAction$1(ScheduledExecutorAsyncRunner.java:119) ~[teku-infrastructure-async-develop.jar:21.11.0+64-gdf02342]
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) [?:?]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) [?:?]
at java.lang.Thread.run(Thread.java:832) [?:?]
Caused by: tech.pegasys.teku.spec.logic.common.statetransition.exceptions.BlockProcessingException: Execution payload parent hash does not match previous execution payload header
at tech.pegasys.teku.spec.logic.versions.merge.block.BlockProcessorMerge.processExecutionPayload(BlockProcessorMerge.java:99) ~[teku-ethereum-spec-develop.jar:21.11.0+64-gdf02342]
at tech.pegasys.teku.spec.logic.versions.merge.block.BlockProcessorMerge.processBlock(BlockProcessorMerge.java:82) ~[teku-ethereum-spec-develop.jar:21.11.0+64-gdf02342]
at tech.pegasys.teku.spec.logic.common.block.AbstractBlockProcessor.lambda$processUnsignedBlock$7(AbstractBlockProcessor.java:301) ~[teku-ethereum-spec-develop.jar:21.11.0+64-gdf02342]
at tech.pegasys.teku.spec.datastructures.state.beaconstate.common.AbstractBeaconState.updated(AbstractBeaconState.java:64) ~[teku-ethereum-spec-develop.jar:21.11.0+64-gdf02342]
at tech.pegasys.teku.spec.logic.common.block.AbstractBlockProcessor.processUnsignedBlock(AbstractBlockProcessor.java:299) ~[teku-ethereum-spec-develop.jar:21.11.0+64-gdf02342]
at tech.pegasys.teku.spec.logic.common.util.BlockProposalUtil.createNewUnsignedBlock(BlockProposalUtil.java:78) ~[teku-ethereum-spec-develop.jar:21.11.0+64-gdf02342]
... 16 more
```

Reply:
one day i'll stop confusing method hitting the validated chain head with the ones taking the optimistic one